lib/iex/lib/iex/helpers.ex

```diff
@@ -1068,7 +1068,17 @@ defmodule IEx.Helpers do
     path = Path.expand(path)
 
     if not optional? or File.exists?(path) do
-      path |> File.read!() |> Code.string_to_quoted!(file: path)
+      if imported_paths = Process.get(:iex_imported_paths) do
+        if path in imported_paths do
+          message = "path #{path} was already imported, circular file imports are not allowed"
```

Comment:
I think the message should say something like "skipping path #{path} because it was already imported" but not sure that'd be clear enough. Any ideas?

Reply:
Maybe?

> "path #{path} was already imported, skipping circular file imports"

What you wrote also sounds good.